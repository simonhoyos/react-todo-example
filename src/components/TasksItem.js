import React from 'react';

export function TaskItem({ id, title, done, handleEdit, handleToggleTask }) {
  return (
    <li>
      {title} - {done ? 'completa' : 'incompleta'}
      <button
        onClick={() => handleEdit(id, title, done)}
        data-testid="edit-button"
      >
        Editar
      </button>
      <button
        onClick={() => handleToggleTask(id)}
        data-testid="toggle-button"
      >
        Completar
      </button>
    </li>
  );
}
