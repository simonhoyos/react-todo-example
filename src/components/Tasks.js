import React from 'react';
import { TaskItem } from './TasksItem';

export function Tasks({ tasks, handleEdit, handleToggleTask }) {
  return (
    <ul>
      {tasks.length > 0 && tasks.map(({ id, title, done }) => {
        return (
          <TaskItem
            key={id}
            id={id}
            title={title}
            done={done}
            handleEdit={handleEdit}
            handleToggleTask={handleToggleTask}
          />
        );
      })}
    </ul>
  );
}
