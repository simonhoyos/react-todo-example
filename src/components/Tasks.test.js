import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Tasks } from './Tasks';
import { TaskItem } from './TasksItem';

const tasksMock = [
  { id: 0, title: 'una tarea', done: true },
  { id: 1, title: 'segunda tarea', done: false },
  { id: 2, title: 'tercera tarea', done: false }
];

describe('<Tasks />', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<Tasks tasks={tasksMock} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render 3 tasks', () => {
    const wrapper = shallow(<Tasks tasks={tasksMock} />);
    const items = wrapper.find(TaskItem);

    expect(items).toHaveLength(3);
  });
});
