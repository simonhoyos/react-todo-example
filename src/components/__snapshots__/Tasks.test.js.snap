// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tasks /> should match snapshot 1`] = `
<ul>
  <TaskItem
    done={true}
    id={0}
    key="0"
    title="una tarea"
  />
  <TaskItem
    done={false}
    id={1}
    key="1"
    title="segunda tarea"
  />
  <TaskItem
    done={false}
    id={2}
    key="2"
    title="tercera tarea"
  />
</ul>
`;
